docs/process/ci_build.md

```diff
@@ -0,0 +1,134 @@
+
+## Testing a Workflow
+
+One nice aspect of Github Actions' integration in Github is that if there is a workflow that is triggered to run on pull requests, and if a pull request includes some modification to that workflow, then the pull request will be run _using the pending new changes to the workflow_.  For example, if an existing workflow (ex: `.github/workflows/build-test.yml`) is configured to run 5 benchmarks, and a pull request is made to add a 6th benchmark to that workflow, then the pull request will run 6 benchmarks, not 5.
+
+This means that most changes to a workflow can be tested in the PR that introduces the changes. The effects are visible to the PR author and reviewers alike.
+
+Less frequently, there might be specific changes that cannot be tested via a PR because they only happen on merges to the main branch `master`. For example, API docs changes and benchmark dashboard changes should only occur on merges to master. In this case, you can use your personal fork of the upstream repo as a testing ground. You can remove branch restrictions from pushing directly to `master` on your fork repo's settings, and you can push the changes you want to test directly on `master`. An alternative strategy for using your fork to do testing is, instead of making feature changes directly on `master`, you can maintain the changes on a feature branch as normal for a "git-flow" workflow. And you can create a special testing branch from the feature branch that maintains an extra commit at its head that loosens restrictions and makes fork-specific configuration changes. For example, when testing fixes to dashboards in `unicode-org/icu4x`, a feature branch `ci-bench-per-component` was created. Then, a testing branch [`ci-bench-per-component-test`](https://github.com/echeran/icu4x/pull/13/commits) was created. Feature changes were made on the feature branch, and the `*-test` branch was always rebased on top of the feature branch in order to maintain fork/testing reconfigurations, such as [allowing `master`-only jobs to run on all branches](https://github.com/echeran/icu4x/pull/13/commits/5e83cb4f97a6629b9bfa330aaeb73b3c523f0a8f), making the [API docs / benchmarks copy destination to be on the personal fork](https://github.com/echeran/icu4x/pull/13/commits/61953d9275d3853d96f43ed95e2ff0e9dc5fdd5f), and adding lots of stdout debugging that gets captured in the logs ([example](https://github.com/echeran/icu4x/pull/13/commits/d6168715ac644b785c920b639cd5238e7a52f806)). The `*-test` branch (`ci-bench-per-component-test`) was pushed to the personal fork after each change to trigger workflow test invocations. The feature branch was largely kept locally until the feature work was done, but the important point was to preserve the invariant that the feature branch (`ci-bench-per-component`) was always in a good state to be turned into a PR on the upstream repo.
```

Comment:
As for "pushing directly to `master` on your fork repo's settings", after testing one then can clean up the fork's master branch with 'git reset --hard HEAD~n' followed by a 'push -f' to the master branch on GitHub. 'n' being the number of recent commits to revert. However, this looks a bit messy, not sure we should recommend it.